### ISSUE TITLE
Allow CREATE SUBSCRIPTION command without needing superuser privilege

### ISSUE THREAD

**Issue by burggraf** [supabase/postgres#1646]
**Edit**: this requires PostgreSQL v17, its deployment is in progress.

Supabase currently supports:
`CREATE PUBLICATION my_publication FOR ALL TABLES;`

but does not support:
`CREATE SUBSCRIPTION my_subscription CONNECTION '<connection_string>' PUBLICATION my_publication;`

The error given is:
`ERROR:  42501: must be superuser to create subscriptions`

In order to be able to use logical replication to replicate individual tables (or all tables in a schema) between Supabase projects, `CREATE SUBSCRIPTION` should be executable by the `postgres` user without `supseruser` privileges, in parity with the `CREATE PUBLICATION` command.

**Comment by ianwoodfill** [comment]
Any update here? This guide is clearly outdated if this is no longer supported: https://supabase.com/docs/guides/database/postgres/setup-replication-external

**Comment by alexandredev3** [comment]
Any updates on this issue? I'm having the same problem

**Comment by frankyw** [comment]
Seems like subscriptions are no longer supported, period.

https://supabase.com/docs/guides/database/postgres/roles-superuser

**Comment by rustamd** [comment]
Looking for this as well, hopefully can get implemented soon.

**Comment by steve-chavez** [comment]
This will come for free once pg 17 is supported.

**Comment by t0mdicks0n** [comment]
I followed the same guide on how to setup replication on Supabase but superuser is needed so the guide is obsolete? At least remove that page on the docs almost 1 year after users in this thread notified you about it?

[https://supabase.com/docs/guides/database/postgres/setup-replication-external](https://supabase.com/docs/guides/database/postgres/setup-replication-external)

**Comment by encima** [comment]
Hey folks,

The guide there is for replicating to an `external` system so the subscription is created on the target database. Creating publications etc are supported without superuser permissions

**Comment by Lanznx** [comment]
any update?

**Comment by encima** [comment]
Yep, this is possible for the `postgres` user as of 17.4 and is rolling out
across regions currently

**Comment by soedirgo** [comment]
Update: PG17 has now been rolled out to all regions. You can do `CREATE SUBSCRIPTION` by upgrading to PG17 via Project Settings > Infrastructure, so closing this.

**Comment by vrijmetse** [comment]
I attempted to create subscription but got this error

`ERROR:  08P01: could not create replication slot "rds_sub": ERROR:  must be superuser or replication role to use logical replication slots in the 'rds' namespace`

**Comment by encima** [comment]
@vrijmetse Are you on Postgres 17?

**Comment by vrijmetse** [comment]
> [@vrijmetse](https://github.com/vrijmetse) Are you on Postgres 17?

Yes 17.6.1.011

**Comment by encima** [comment]
Ah, that error looks like a non supabase issue? You can create a normal subscription but not one for the `RDS` namespace

EDIT: 
can confirm this is not a Supabase issue. The slot in the other DB may need superuser perms to create. You can precreate the slot and specify or you can use `rds_superuser`
